### PR TITLE
New page: overflow values

### PR DESCRIPTION
**Pull request by estelle**
The `<overflow>` values are used in multiple properties. Instead of defining the values in each location, we can provide a deep dive in one location and refer to this page when using the value type.

### CODE REVIEW

**Comment by dipikabh** [comment]
Thanks, @estelle. I'll start reviewing this tomorrow.

File: files/en-us/web/css/overflow_value/index.md [hunk 1]
```diff
@@ -0,0 +1,249 @@
...
+- `scroll`
+  - : Overflow content is clipped at the element's padding box, and overflow content can be scrolled into view using scroll bars. User agents display scroll bars in both horizontal and vertical directions if only one value is set, whether or not any content is overflowing or clipped. The use of this keyword, therefore, can prevent scroll bars from appearing and disappearing as content changes. Printers may still print overflowing content. The element box on which this value is set is a {{glossary("scroll container")}}.
+- `auto`
+  - : Overflow content is clipped at the element's padding box, and overflow content can be scrolled into view. Unlike `scroll`, user agents display scroll bars _only if_ the content is overflowing and hide scroll bars by default. If content fits inside the element's padding box, it looks the same as with `visible` but still establishes a new formatting context. The element box on which this value is set is a {{glossary("scroll container")}}.
+
+> **Note:** The keyword value `overlay` is a legacy value alias for `auto`. With `overlay`, the scroll bars are drawn on top of the content instead of taking up space.
```
Comment (yarusome) [inline files/en-us/web/css/overflow_value/index.md:35]: The `overlay` value *should* strictly behave the same as `auto` now per https://github.com/w3c/csswg-drafts/issues/8063#issuecomment-1431620007. However, according to [Firefox Bug 1817189](https://bugzilla.mozilla.org/show_bug.cgi?id=1817189), [WebKit Changeset 236341](https://trac.webkit.org/changeset/236341/webkit), and [Chrome Status](https://chromestatus.com/feature/5194091479957504), it seems that Chrome alone retains the legacy behavior.

File: files/en-us/web/css/overflow_value/index.md [hunk 2]
```diff
@@ -0,0 +1,249 @@
...
+}
+
+pre:nth-of-type(5) {
+  overflow: clip;
+  overflow: overlay;
+  overflow-clip-margin: 3em;
+}
+pre:nth-of-type(5)::before {
+  content: "overlay (or clip if not supported): ";
+}
```
Comment (yarusome) [inline files/en-us/web/css/overflow_value/index.md:220]: …and this block, along with one of the `<p>` elements above, could be removed.

File: files/en-us/web/css/overflow_value/index.md (hunk 1 above)
Comment (dipikabh) [reply files/en-us/web/css/overflow_value/index.md:35]: In testing on Firefox 112, Chrome 112, and Safari 16, this behavior of `overlay`, where the scroll bars are drawn on top of the content, is found to be true. So we're calling out that difference from `auto`.

File: files/en-us/web/css/overflow_value/index.md [hunk 3]
```diff
@@ -0,0 +1,247 @@
+---
+title: <overflow>
+slug: Web/CSS/overflow_value
+page-type: css-type
+browser-compat: css.types.overflow
+---
+
+{{CSSRef}}
+
+The **`<overflow>`** {{glossary("enumerated")}} values is a set of keyword values used in the {{cssxref("overflow-x")}}, {{cssxref("overflow-y")}}, {{cssxref("overflow-inline")}} and {{cssxref("overflow-block")}} properties, as well as the {{cssxref("overflow")}} shorthand property. These properties apply to block containers, flex containers, and grid containers.
```
Comment (estelle) [inline files/en-us/web/css/overflow_value/index.md:10]:
```suggestion
The **`<overflow>`** {{glossary("enumerated")}} value is a set of keyword values used in the {{cssxref("overflow-x")}}, {{cssxref("overflow-y")}}, {{cssxref("overflow-inline")}} and {{cssxref("overflow-block")}} properties, as well as the {{cssxref("overflow")}} shorthand property. These properties apply to block containers, flex containers, and grid containers.
```

**Comment by estelle** [comment]
> Hi @estelle, the [Overflow specification](https://w3c.github.io/csswg-drafts/css-overflow/) does not define `<overflow>` as a data type. Would it be correct to document it as one?

It doesn't defined it as a data type. Similar to `<line-style>`, it's an enumerated value type, so let's go with that, "enumerated value type", categorized as a data-type for page types, but not listing the term.


> As an aside - I don't think there is any other "_value" data type page (overflow_value used in the slug for the new page) in the "Types" sidebar grouping.

as `overflow` is a property, i couldn't use that as the slug. i grabbed the slug idea from `color_value` as `color` is also a property.

File: files/en-us/web/css/overflow_value/index.md [hunk 4]
```diff
@@ -0,0 +1,247 @@
+---
+title: <overflow>
+slug: Web/CSS/overflow_value
+page-type: css-type
+browser-compat: css.types.overflow
+---
+
+{{CSSRef}}
+
+The **`<overflow>`** {{glossary("enumerated")}} value type defines the set of overflow keyword values used in the {{cssxref("overflow-x")}}, {{cssxref("overflow-y")}}, {{cssxref("overflow-inline")}} and {{cssxref("overflow-block")}} properties, as well as the {{cssxref("overflow")}} shorthand property. These properties apply to block containers, flex containers, and grid containers.
```
Comment (dipikabh) [inline files/en-us/web/css/overflow_value/index.md:10]:
```suggestion
The **`<overflow>`** {{glossary("enumerated")}} value type represents the keyword values for the {{cssxref("overflow-block")}}, {{cssxref("overflow-inline")}}, {{cssxref("overflow-x")}}, and {{cssxref("overflow-y")}} longhand properties and the {{cssxref("overflow")}} shorthand property. These properties apply to block containers, flex containers, and grid containers.
```

Does the [Enumerated](https://developer.mozilla.org/en-US/docs/Glossary/Enumerated) glossary page need to be updated to include a mention of enumerated values in CSS?

File: files/en-us/web/css/overflow_value/index.md [hunk 5]
```diff
@@ -0,0 +1,247 @@
...
+
+## Syntax
+
+```css
+<overflow> = visible | hidden | clip | scroll | auto
+```
+
+## Values
+
+The `<overflow>` data type is specified using one of the values listed below.
```
Comment (dipikabh) [inline files/en-us/web/css/overflow_value/index.md:20]:
```suggestion
The `<overflow>` enumerated value type is specified using one of the values listed below.
```

File: files/en-us/web/css/overflow_value/index.md [hunk 6]
```diff
@@ -0,0 +1,247 @@
...
+- `scroll`
+  - : Overflow content is clipped at the element's padding box, and overflow content can be scrolled into view using scroll bars. User agents display scroll bars in both horizontal and vertical directions if only one value is set, whether or not any content is overflowing or clipped. The use of this keyword, therefore, can prevent scroll bars from appearing and disappearing as content changes. Printers may still print overflowing content. The element box on which this value is set is a {{glossary("scroll container")}}.
+- `auto`
+  - : Overflow content is clipped at the element's padding box, and overflow content can be scrolled into view. Unlike `scroll`, user agents display scroll bars _only if_ the content is overflowing and hide scroll bars by default. If content fits inside the element's padding box, it looks the same as with `visible` but still establishes a new formatting context. The element box on which this value is set is a {{glossary("scroll container")}}.
```
Comment (dipikabh) [inline files/en-us/web/css/overflow_value/index.md:31]:
```suggestion
- `visible`
  - : Overflow content is not clipped and may be visible outside the element's padding box. The element box is not a {{glossary("scroll container")}}. This is the default value for all the properties that have the `<overflow>` enumerated value type.
- `hidden`
  - : Overflow content is clipped at the element's padding box. There are no scroll bars, and the clipped content is not visible (i.e., clipped content is hidden), but the content still exists. User agents do not add scrollbars and also do not allow users to view the content outside the clipped region by actions such as dragging on a touch screen or using the scroll wheel on a mouse. The content _can_ be scrolled programmatically (for example, by setting the value of the {{domxref("Element.scrollLeft", "scrollLeft")}} property or the {{domxref("Element.scrollTo", "scrollTo()")}} method). The content can also be scrolled via keyboard interaction; arrows enable scrolling through the content and tabbing to a focusable element within the hidden content enables scrolling the focused element into view. The element box on which this value is set is a scroll container.
- `clip`
  - : Overflow content is clipped at the element's _overflow clip edge_ that is defined using the [`overflow-clip-margin`](/en-US/docs/Web/CSS/overflow-clip-margin) property. As a result, content overflows the element's padding box by the {{cssxref("&lt;length&gt;")}} value of `overflow-clip-margin` or by `0px` if not set. Overflow content outside the clipped region is not visible, user agents do not add a scrollbar, and programmatic scrolling is also not supported. No new [formatting context](/en-US/docs/Web/Guide/CSS/Block_formatting_context) is created.
- `scroll`
  - : Overflow content is clipped at the element's padding box, and overflow content can be scrolled into view using scrollbars. User agents display scrollbars in both horizontal and vertical directions if only one value is set, whether or not any content is overflowing or clipped. The use of this keyword value, therefore, can prevent scrollbars from appearing and disappearing as content changes. Printers may still print overflowing content. The element box on which this value is set is a scroll container.
- `auto`
  - : Overflow content is clipped at the element's padding box, and overflow content can be scrolled into view. Unlike `scroll`, user agents display scrollbars _only if_ the content is overflowing and hide scrollbars by default. If content fits inside the element's padding box, it looks the same as with `visible` but still establishes a new formatting context. The element box on which this value is set is a scroll container.
```
- I had added a space in "scroll bars" but after looking at some articles, I am now leaning towards the one word "scrollbar".
- In `scroll` description, we say "User agents display scrollbars in both horizontal and vertical directions if only one value is set" but we have not mentioned on this page that properties can accept two values, one for each direction.

File: files/en-us/web/css/overflow_value/index.md [hunk 7]
```diff
@@ -0,0 +1,247 @@
...
+
+> **Note:** The keyword value `overlay` is a legacy value alias for `auto`. With `overlay`, the scroll bars are drawn on top of the content instead of taking up space.
+
+## Examples
+
+This example shows all the `<overflow>` values as values of the CSS {{cssxref("overflow")}} property.
```
Comment (dipikabh) [inline files/en-us/web/css/overflow_value/index.md:37]:
```suggestion
This example demos all the `<overflow>` enumerated values for the {{cssxref("overflow")}} property.
```

File: files/en-us/web/css/overflow_value/index.md [hunk 8]
```diff
@@ -0,0 +1,247 @@
...
+
+### HTML
+
+We include lyrics, using the HTML {{HTMLELement("pre")}} element for preformatted content. The same HTML is repeated multiple times. A link is included to enable testing the effects of keyboard focus on overflow and scroll behaviors.
```
Comment (dipikabh) [inline files/en-us/web/css/overflow_value/index.md:41]:
```suggestion
The HTML in this example contains some lyrics within the {{HTMLELement("pre")}} element. The HTML also contains a link text to enable testing the effects of keyboard focus on overflow and scroll behaviors. The same HTML code is repeated multiple times to show the effect of each `<overflow>` enumerated value. 
```
tabbing on link will not work on Firefox

File: files/en-us/web/css/overflow_value/index.md [hunk 9]
```diff
@@ -0,0 +1,247 @@
...
+
+<a href="#">Rubber Duckie</a>, you're so fine
+And I'm lucky that you're mine
+Rubber Duckie, I'm awfully fond of you
+</pre>
+```
+
+### CSS
+
+We define the size of `<pre>` to ensure the content overflows its container in both the inline and block directions. We set a different `<overflow>` value in each example, and include generated content to help you identify which overflow value is being demonstrated:
```
Comment (dipikabh) [inline files/en-us/web/css/overflow_value/index.md:157]:
```suggestion
For the purpose of demonstration, the size of the `<pre>` element box has been defined to ensure that the content overflows its container in both the inline and block directions. A different `<overflow>` value is set for each of the repeating `<pre>` elements. For the `clip` value demonstration, a {{CSSXref("overflow-clip-margin")}} has been added.
```

File: files/en-us/web/css/overflow_value/index.md [hunk 10]
```diff
@@ -0,0 +1,247 @@
...
+pre:nth-of-type(6) {
+  overflow: visible;
+}
+pre:nth-of-type(6)::before {
+  content: "visible: ";
+}
+```
+
+We included {{CSSXref("overflow-clip-margin")}}, which only has an effect when used in conjunction with the `clip` value to demonstrate the clip effect.
+
```
Comment (dipikabh) [inline files/en-us/web/css/overflow_value/index.md:229]:
```suggestion

```
Moved this up along with CSS code description

File: files/en-us/web/css/overflow_value/index.md [hunk 11]
```diff
@@ -0,0 +1,247 @@
...
+### Results
```
Comment (dipikabh) [inline files/en-us/web/css/overflow_value/index.md:230]:
```suggestion
### Result
```

File: files/en-us/web/css/overflow_value/index.md [hunk 12]
```diff
@@ -0,0 +1,247 @@
...
+
+{{EmbedLiveSample("Examples", "500", "600")}}
+
+A link is included in each `<pre>` to demonstrate the effects of keyboard focus on overflow and scroll behaviors. Try tabbing thru all the links in the example. Notice that the `clip` example does not create a {{glossary("scroll container")}}, and the link does not come into view when the link is focused. The `visible` value, which has the link always in view, is also not a scroll container.
```
Comment (dipikabh) [inline files/en-us/web/css/overflow_value/index.md:234]:
```suggestion
To see the effect of keyboard focus on overflow and scroll behaviors, try tabbing through all the links in the example. Notice that the `clip` box does not create a scroll container, and the link does not come into view when the link is focused. The `visible` value, which has the link always in view, is also not a scroll container.
```

File: files/en-us/web/css/overflow_value/index.md [hunk 13]
```diff
@@ -0,0 +1,247 @@
...
+pre:nth-of-type(6)::before {
+  content: "visible: ";
+}
+```
+
+We included {{CSSXref("overflow-clip-margin")}}, which only has an effect when used in conjunction with the `clip` value to demonstrate the clip effect.
+
+### Results
+
+{{EmbedLiveSample("Examples", "500", "600")}}
```
Comment (dipikabh) [inline files/en-us/web/css/overflow_value/index.md:232]: You might want to increase the length of the iframe here to contain all the examples

File: files/en-us/web/css/overflow_value/index.md [hunk 14]
```diff
@@ -0,0 +1,247 @@
...
+
+{{Specifications}}
+
+## Browser compatibility
+
+{{Compat}}
+
+## See also
+
+- Properties that use this data type: {{cssxref("overflow-x")}}, {{cssxref("overflow-y")}}, {{cssxref("overflow-inline")}}, {{cssxref("overflow-block")}} and {{cssxref("overflow")}}
```
Comment (dipikabh) [inline files/en-us/web/css/overflow_value/index.md:246]:
```suggestion

```
The list has already been linked at the top of the page

**Review by dipikabh — CHANGES_REQUESTED** [review]
Getting around to the idea of having a page for enumerated value type! Suggested some edits and rephrasing

File: files/en-us/web/css/overflow_value/index.md [hunk 15]
```diff
@@ -0,0 +1,247 @@
+---
+title: <overflow>
+slug: Web/CSS/overflow_value
+page-type: css-type
+browser-compat: css.types.overflow
```
[the remark this answers is not in the document]
Comment (estelle) [reply files/en-us/web/css/overflow_value/index.md:5]: we're pretending this is a regular data type for now: it's just enumerated rather than functional or numeric like the other ones

File: files/en-us/web/css/overflow_value/index.md (hunk 4 above)
Comment (estelle) [reply files/en-us/web/css/overflow_value/index.md:10]: Good call. I think we're ok with the current definition: "In computer science, an enumerated type is a data type consisting of a limited set of named values.". It'ss enough of a definition for CSS. Enumerated in HTML and JS has special meaning in addition to that line.

File: files/en-us/web/css/overflow_value/index.md [hunk 16]
```diff
@@ -0,0 +1,247 @@
+---
+title: <overflow>
+slug: Web/CSS/overflow_value
+page-type: css-type
```
[the remark this answers is not in the document]
Comment (estelle) [reply files/en-us/web/css/overflow_value/index.md:4]: for now, we are adding this as a regular value type. We can change if we think it doesn't work live. 
For the slug, other CSS data types use "value" not "values.

File: files/en-us/web/css/overflow_value/index.md [hunk 17]
```diff
@@ -0,0 +1,245 @@
...
+  overflow: visible;
+}
+pre:nth-of-type(6)::before {
+  content: "visible: ";
+}
+```
+
+### Result
+
+{{EmbedLiveSample("Examples", "500", "600")}}
```
Comment (estelle) [inline files/en-us/web/css/overflow_value/index.md:230]:
```suggestion
{{EmbedLiveSample("Examples", "500", "800")}}
```

**Review by dipikabh — APPROVED** [review]
Thank you for the updates!
